sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredentialBuilder.java

```diff
@@ -29,6 +29,31 @@ public DeviceCodeCredentialBuilder challengeConsumer(
         return this;
     }
 
+    /**
+     * Sets whether to use an unprotected file specified by <code>cacheFileLocation()</code> instead of
+     * Gnome keyring on Linux. This is false by default.
+     *
+     * @param allowUnencryptedCache whether to use an unprotected file for cache storage.
+     *
+     * @return An updated instance of this builder with the unprotected token cache setting set as specified.
+     */
+    public DeviceCodeCredentialBuilder allowUnencryptedCache(boolean allowUnencryptedCache) {
+        this.identityClientOptions.allowUnencryptedCache(allowUnencryptedCache);
+        return this;
+    }
+
+    /**
+     * Sets whether to enable using the shared token cache. This is disabled by default.
+     *
+     * @param enabled whether to enabled using the shared token cache.
+     *
+     * @return An updated instance of this builder with if the shared token cache enabled specified.
+     */
+    public DeviceCodeCredentialBuilder enableSharedTokenCache(boolean enabled) {
```

Comment:
In .NET and python we named this option EnablePersistentCache.

Reply:
updated